core/sequencer/server.go

```diff
@@ -274,14 +274,16 @@ func (s *Server) DefineRevisions(ctx context.Context,
 	if err != nil {
 		return nil, err
 	}
-	outstanding := []int64{}
-	for rev := int64(latestMapRoot.Revision) + 1; rev <= highestRev; rev++ {
-		outstanding = append(outstanding, rev)
-	}
 
-	// Don't create new revisions if there are ones waiting to be applied.
-	if len(outstanding) > 0 {
-		return &spb.DefineRevisionsResponse{OutstandingRevisions: outstanding}, nil
+	resp := &spb.DefineRevisionsResponse{
+		MapRevision: int64(latestMapRoot.Revision),
+		OldDefined:  highestRev,
+		NewDefined:  highestRev,
+	}
+	// Don't create new revisions if there are ones waiting to be applied, or the
+	// highest defined revision is lagging behind for some reason.
+	if resp.OldDefined != resp.MapRevision {
```

Comment:
Should this logic be with respect to NewDefined?

Reply:
Now irrelevant.